.github/workflows/nightly-test-integrations.yml

```diff
@@ -0,0 +1,314 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+name: Nightly test-integrations
```

Comment:
**thought:** If we want nightly test runs for the release branches also, then you will need separate cron workflows similar to this and will need to set the release branch on checkout.

This is because Github Actions run all of the workflows off of main. You do this in the code checkout step.

```
     - name: Checkout code
        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
        with:
            ref: ${{ inputs.branch }}
 ```